lib/deterministicmnlist/QuorumEntry.js

```diff
@@ -0,0 +1,269 @@
+  bufferWriter.writeUInt8(this.llmqType);
+  bufferWriter.write(Buffer.from(this.quorumHash, 'hex').reverse());
+  bufferWriter.writeVarintNum(this.getParams().size);
+  bufferWriter.write(Buffer.from(this.validMembers, 'hex'));
+  bufferWriter.write(Buffer.from(this.quorumPublicKey, 'hex'));
+  bufferWriter.write(Buffer.from(this.quorumVvecHash, 'hex').reverse());
+
+  return Hash.sha256sha256(bufferWriter.toBuffer()).reverse();
+};
+
+/**
+ * Create SMLQuorumEntry from an object
+ * @param {SMLQuorumEntry} obj
+ * @return {QuorumEntry}
+ */
+QuorumEntry.fromObject = function fromObject(obj) {
+  var SMLQuorumEntry = new QuorumEntry();
+  SMLQuorumEntry.version = obj.version;
+  SMLQuorumEntry.llmqType = obj.llmqType;
+  SMLQuorumEntry.quorumHash = obj.quorumHash;
+  SMLQuorumEntry.signersCount = obj.signersCount;
+  SMLQuorumEntry.signers = obj.signers;
+  SMLQuorumEntry.validMembersCount = obj.validMembersCount;
+  SMLQuorumEntry.validMembers = obj.validMembers;
+  SMLQuorumEntry.quorumPublicKey = obj.quorumPublicKey;
+  SMLQuorumEntry.quorumVvecHash = obj.quorumVvecHash;
+  SMLQuorumEntry.quorumSig = obj.quorumSig;
+  SMLQuorumEntry.membersSig = obj.membersSig;
+
+  SMLQuorumEntry.validate();
+  return SMLQuorumEntry;
+};
+
+QuorumEntry.prototype.validate = function validate() {
+  $.checkArgument(utils.isUnsignedInteger(this.version), 'Expect version to be an unsigned integer');
+  $.checkArgument(utils.isUnsignedInteger(this.llmqType), 'Expect llmqType to be an unsigned integer');
+  $.checkArgument(isSha256(this.quorumHash), 'Expected quorumHash to be a sha256 hex string');
+  $.checkArgument(utils.isHexaString(this.signers), 'Expect signers to be a hex string');
+  $.checkArgument(utils.isHexaString(this.validMembers), 'Expect validMembers to be a hex string');
+  $.checkArgument(isHexStringOfSize(this.quorumPublicKey, BLS_PUBLIC_KEY_SIZE * 2), 'Expected quorumPublicKey to be a bls pubkey');
+  $.checkArgument(isHexStringOfSize(this.quorumVvecHash, SHA256_HASH_SIZE * 2), `Expected quorumVvecHash to be a hex string of size ${SHA256_HASH_SIZE}`);
+  $.checkArgument(isHexStringOfSize(this.quorumSig, BLS_SIGNATURE_SIZE * 2), 'Expected quorumSig to be a bls signature');
+  $.checkArgument(isHexStringOfSize(this.membersSig, BLS_SIGNATURE_SIZE * 2), 'Expected membersSig to be a bls signature');
+};
+
+QuorumEntry.prototype.toObject = function toObject() {
+  return {
+    version: this.version,
+    llmqType: this.llmqType,
+    quorumHash: this.quorumHash,
+    signersCount: this.signersCount,
+    signers: this.signers,
+    validMembersCount: this.validMembersCount,
+    validMembers: this.validMembers,
+    quorumPublicKey: this.quorumPublicKey,
+    quorumVvecHash: this.quorumVvecHash,
+    quorumSig: this.quorumSig,
+    membersSig: this.membersSig,
+  };
+};
+
+/**
+ * @return {number}
+ */
+QuorumEntry.prototype.getParams = function getParams() {
+  var params = {};
+  switch (this.llmqType) {
+    case 1:
```

Comment:
Let's use constants here instead of magic numbers?

Reply:
Done. Thank you!